### DISCUSSION TITLE
MySQL connecition timeout

### DISCUSSION THREAD

**Discussion by legaljustus** [Budibase/budibase#6397] [Help]
Hi!

I have been working with budibase (selfhosted on AWS) for a while, big fan! For my most recent project I needed to connect a mySQL database (we normally use postgeSQL). In this MySQL database there are tables with 100k plus records of about 10 text and timestamp columns. I have a search functionality for this table setup using a dataprovider with a filter coupled to a form field (see config below)

![image](https://user-images.githubusercontent.com/83215912/174617493-78a7f425-1bae-46d6-b37f-40f7eff8160a.png)

but now whenever i do multiple searches in a short timespan i get the following error: 
![image](https://user-images.githubusercontent.com/83215912/174618426-8f64efa0-d1f3-4f8c-86b4-2f0d766bf9cd.png)

Also when i look at my larger tables in the data tab, it does't preview any records: 
![image](https://user-images.githubusercontent.com/83215912/174618618-20b5981a-3186-4c24-86cf-a321ba701abe.png)

On the sql server side I don't see that much usage of mysql. So does this maybe have to do with the timeout configuration or something?

Any help would be greatly appreciated

**Comment by melohagan** [comment]
Hi @legaljustus 

The SQL Server implementation uses a connection pool, whereas the MySQL implementation creates a new connection every-time.

That could be the issue! 

There's a feature request open to support pooling for "plus datasources" which would include MySQL. https://github.com/Budibase/budibase/issues/6130

**Reply by legaljustus** [reply]
Hi! sorry for the late reply, was on a short break!

I checked in with my devops guy and he sent me that it's probably not the problem as the SQL connections are not really changing that often (looking at the AWS graphs). What do you mean by every-time? Every-time the query changes, so for example if i change a single character in a search field? Because then I think connection pooling is needed to make the mySQL connection work properly. I already gave the feature request a thumbs up!
